contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -603,6 +651,90 @@ def _find_timecode_track_start(track):
     )
 
 
+def _find_mastermob_for_sourceclip(aaf_sourceclip):
+    """
+    For a given soure clip, find the related masterMob
+    """
+
+    # If the mobId of the sourceclip is a mastermob, just return that, we are done.
+    if isinstance(aaf_sourceclip.mob, aaf2.mobs.MasterMob):
+        return aaf_sourceclip.mob, None
+
+    # There are cases where a composition mob is used as an indirection
+    # to the mastermob. In that case the SourceClip points to a
+    # CompositionMob instead of a MasterMob. Drill down into the CompositionMob
+    # to find the MasterMob related to this SourceClip
+    return _get_master_mob_from_source_composition(aaf_sourceclip.mob)
+
+
+def _get_master_mob_from_source_composition(compositionMob):
+    """
+    This code covers two special cases:
+    if the passed in source-clip-mob is a composition, drill down
+    and try to find the master mob in that composition.
+
+    Also, there seems to be a workflow where metadata, specifically UserComments
+    are shared between SourceClips via a CompositionMob, in which case there are
+    no UserComments on the MasterMob (as we expect in the default case)
+
+    So if we find UserComments on the Composition but not on the MasterMob, we
+    return that metadata, so it can be added to the clip (instead of the
+    master mob UserComments)
+
+    """
+
+    # If not a composition, we can't discover anything
+    if not isinstance(compositionMob, aaf2.mobs.CompositionMob):
+        return None, None
+
+    compositionMetadata = _get_composition_user_comments(compositionMob)
+
+    # Iterate over the TimelineMobSlots and extract any source_clips we find.
+    source_clips = []
+    for slot in compositionMob.slots:
+        if isinstance(slot, aaf2.mobslots.TimelineMobSlot):
+            if isinstance(slot.segment, aaf2.components.SourceClip):
+                source_clips.append(slot.segment)
+
+    # No source clips, no master mob. But we still want to return
+    # the composition metadata. If there is another mastermob found 'upstream',
+    # but it does not have any UserComments metadata, we still want to use
+    # the CompositionMob's metadata.
+    if not source_clips:
+        return None, compositionMetadata
+
+    # sschulze 06/22/20 - Only expect one source clip for this case.
```

Comment:
We tend to avoid leaving people's names/logins in the open source code base.

Reply:
sorry, that was from my tools-tree change. I'll remove that.